### ISSUE TITLE
Add tiktok to social media?

### ISSUE THREAD

**Issue by xcasxcursex** [StevenBlack/hosts#1279]
Sorry, I read the front page and the suggestions on how to add sites to the lists, but it's not clear which of the various lists hosts the social media category.

Also, I am honestly not entirely sure what domains need to be blocked. This repo seems to have a full list: https://github.com/d43m0nhLInt3r/socialblocklists
That repo has other useful lists (eg whatsapp) So I thought perhaps you may prefer to just add it as a source?

Anyway, clearly, I'm not sure where to go from here. I was just surprised when an embedded tiktok video actually loaded on my PC ;)

**Comment by XhmikosR** [comment]
There's also https://github.com/llacb47/mischosts/blob/master/tiktok-hosts from @llacb47

**Comment by StevenBlack** [comment]
Are there REALLY 94 domains to block for TicToc?  🧐 That’s pretty bad.

**Comment by CyanoTex** [comment]
> 
> 
> Are there REALLY 94 domains to block for TikTok? 🧐 That’s pretty bad.

According to NextDNS' metadata repo, parentalcontrol/services/tiktok, these are enough to block TikTok:
```
tiktok.com
tiktokcdn.com
musical.ly
muscdn.com
tiktokv.com
tiktokcdn-com.akamaized.net
```
From what I can tell, the repo Mikos gave has a few domains Unified already blocks, but aside from that, I can't ascertain if TikTok really has that many domains.

Regardless, it would be a welcome addition.

**Comment by llacb47** [comment]
The donains in my list are just tracking domains and are not intended for someone who wants to disable access to tiktok. So not a good fit for the social media list.

**Comment by StevenBlack** [comment]
Ping @sinfonietta!

**Comment by StevenBlack** [comment]
Nice!  Thanks @sinfonietta.

Closing.

**Comment by xcasxcursex** [comment]
Fantastic, thanks!

**Comment by llacb47** [comment]
This commit https://github.com/Sinfonietta/hostfiles/commit/f63240af51dceeed3a783ea62d975c29fbb27fd3 is wholly inadequate for most users of this list. 


It is fine when utilizing this list on Adguard Home or NextDNS, as these platforms block subdomains. However, on PiHole and as an actual host file, this will not actually block access to the tiktok app. 
I would suggest my list (😉): https://github.com/llacb47/mischosts/blob/master/social/tiktok-block

**Comment by StevenBlack** [comment]
Here's my thinking on the **extensions** to this list.  Porn, Gambling, and Social Media are extensions.

All the main use cases of extensions are disuaisive.  

The archetypal target for the Gambling extension is the problem gambler who needs to get his shit together, and not gamble online anymore.  Or at least, not at work, or not on specific machines.

One archetypal target for the Social Media extension is someone who wants off that habitual train.  A secondary target for the Social Media extension is parents who wants to limit kid's exposure.   Or an employer that wants to keep that stuff out of the office.

The same is true of the Porn extension.  It's main role is to help someone who's hooked on that.  Secondarily, it's for parents to keep that stuff out of their home.  Or an employer that wants to keep that stuff out of the office.

Now **all that said**, we're talking about fucking TicToc, here.  It's helpful, I think, to keep a proper perspective on all this.

Should we block what we can?  Yes!  Should we go crazy and block 165,000 domains for porn?  Some porn lists have 10^6 magnitude domains for porn.  That would be a factor of 8 more than we carry.  Our list stops all casual porn browsing, and lots more.  For the extra factor of 8 or 10, you only get small, arguably negligible, incremental coverage.

The goal isn't to crush TicToc.  The goal is to help users who want to stop Social Media from interrupting their life.  Do we do that?  If yes, I'm fine with it.  If not, let's fix it.  but `obscure.subdomain.tiktoc.com` is only a problem if it's a common pathway into TicToc.

**Comment by xcasxcursex** [comment]
TL;DR skip to the last line :)

Thanks for checking back in @llacb47  . I'm confused, I thought that you said your list was not appropriate?

Perhaps it will help to explain my thoughts on what should be blocked as part of the social media list (and this goes for any other; gambling, porn, etc).... Whether it's blocking the app or blocking tracking or blocking the website, it's the entire thing, all of them, that I am trying to block. 

At least to me, and I'm sure to many others as well, blocking social media, is not so that I can't use social media - it's so that social media can't use me.

From a practical perspective, if one blocks all aspects of a given category (in this case social media) then they include the others, but the inverse is not true. Accordinlgy, only reason not to block trackers and all other aspects (phone apps, whatever) if if one wants to block the experience to the end user but explicitly allow the experience for the vendor (ie the social media network)
.
.
.
As soon as I finished that sentence, @StevenBlack posted and made it clear that is precisely his intention. Now Steven this is YOUR repo and I do respect that, but perhaps it is worthy of considering that for some users, the goal IS to 'crush tictoc' - in the defensive sense, not the offensive, because tictoc will happily crush them (something more apparent with addictions like gambling and porn but no less true of social media). Especially in the case of social media, where damage is done by controlling end-user behaviour via advertising powered by datamining, the tracking is often more damaging than the visitation of the site for social purposes and the negative side effects that exist there.

You're right though, there are diminishing returns here, as the more you add to the list, the less effect it has, and of course there are also performance penalties to consider.

Again, this is your repo, and I'm not trying to tell you how to run it :) However, any block list is always going to keep growing (eg this thread), and will always have different intentions (eg blocking porn addictions or preventing tracking), so I think some consideration of how to handle that, is worthwhile.

Since I'd like to present a possible solution now that I've presented a problem:  Perhaps a tracking 'extension' to the list, might be a good way to deal with it? That seems appropriate given that there's little point in blocking one dataminer (social media) and allowing the others (porn etc) since they all share their data anyway.

**Comment by StevenBlack** [comment]
@xcasxcursex another angle is, what distinguishes TikTok from YouTube?

We don't block `youtube.com` or `m.youtube.com`.  Yet there are comment threads in YouTube.  It's arguably a very strong player in the social media space.

So why don't we block youtube in our Social Media extension?

Because when you need youtube, you need it bad — to fix something, for example.  YouTube is a major player in education, at all levels.

We don't block YouTube's basic functionality because we aren't dogmatic about all this.  We assess, we talk about it, and we draw a line that makes sense.  We can always re-draw the lines.

**Comment by xcasxcursex** [comment]
What distinguishes them? I dunno, they're spelled differently, that's about it ;)

You're absolutely right and you made great examples - I block comment threads in youtube (CSS display:none on the entire section via ublock) but unfortunately it's a platform I need to use so I can't block the whole thing. Reddit, I have an exclusion in your social extension for, because very often it's a hub for information on technical things (eg firefox nightly builds). As you say, I re-draw the lines to make sense :) I would argue strongly that youtube.com _should_ be in the social media list.

I'm just saying, that blocking tracking does make sense for some people.

It's much easier, form the user's perspective, to whitelist certain sites (like YouTube), or not install the blacklist for a certain category (like tracking). than it is for them to collate a list of tracking sites (a struggle with which I'm sure you are familiar) or fight an addiction to the comment section. If ease of use is a concern (such as the commonality of whitelisting youtube) then a list of commonly whitelisted sites could also be an extension.

I should have foreseen, that blocking tiktok should be such a can of worms ;) 

Just for what it's worth - the reason that this issue exists, is that I like to block embedded third-party videos to avoid tracking. I have never, and probably never will, put tiktok.com in my URL bar and hit enter. I'm pretty sure that firefox's privacy and tracking options block embedded videos already for most sites, so I usually just get a grey rectangle, and if I really want to see the video, I open it in another tab. I was surprised that firefox failed to block it at first, and then it hit me - why aren't my hostsfiles blocking this, too? So, this issue was created. I thought I should tell this story to give some perspective on another use-case for your excellent list.

**Comment by llacb47** [comment]
> I'm confused, I thought that you said your list was not appropriate?

I created a social version of my tiktok list. 

> The goal isn't to crush TikTok. The goal is to help users who want to stop Social Media from interrupting their life. Do we do that? If yes, I'm fine with it. If not, let's fix it.

@StevenBlack If you want, I can create a more light-weight abbreviated version of my list. - no CNAMEs, no obscure domains. Only domains needed to impede access to tiktok websites and apps.
